cmake/linux/package_linux.sh.in

```diff
@@ -103,7 +103,7 @@ mv "${APPDIR}usr/bin/lmms" "${APPDIR}usr/bin/lmms.real"
 cat >"${APPDIR}usr/bin/lmms" <<EOL
 #!/usr/bin/env bash
 DIR="\$( cd "\$( dirname "\${BASH_SOURCE[0]}" )" && pwd )"
-export LD_LIBRARY_PATH=\$DIR/usr/lib/:\$DIR/usr/lib/lmms:\$LD_LIBRARY_PATH
+export LD_LIBRARY_PATH=\$DIR/usr/lib/:\$DIR/usr/lib/lmms:/usr/lib/carla/:/usr/lib64/carla/:\$LD_LIBRARY_PATH
```

Comment:
@falkTX so I complied Carla from source on Fedora 26 and it puts the library into `/usr/local/lib/carla/`. New `carla.cpp` logic in LMMS works fine however these AppImages change linking so like the above diff, we have a need for the same `LD_LIBRARY_PATH` manipulation above for it to be found properly.

1. From your experience, is `/usr/local/lib/carla/` common enough to just add into our launcher?
2. I could try to hack something like `which carla` into the script, grab the prefix and add `lib` after it, but it won't handle ycollet's issue unless we blindly add `lib64` as well to `LD_LIBRARY_PATH` as well.  Thoughts welcome here.

Reply:
On systems that use lib64, they will have that as part of the default library search path, so not sure if this is going to be an issue.
You can use which carla, sure. that even detects if carla is installed on the system and skip setting any stuff for carla (which can be a good thing).

I believe carla on lmms sets linker rpath though, is that not the same for the AppImage build?
(rpath meaning the full path to the carla library is hardcoded)

Reply:
> On systems that use lib64, they will have that as part of the default library search path, so not sure if this is going to be an issue.

That's true for `/usr/lib/` and friends, but not subfolders.  I'm not sure how else to predict a foreign systems' `LD_LIBRARY_PATH`.  From my previous research, unix doesn't really offer this information to a shell script.

Reply:
Ah right.. carla is not on libdir directly..
There is /etc/ld.so.conf.d/ and ldconfig, but using this seems hacky as well.

I can add a script to carla which prints the libdir.
Failing to find/call that script means carla is not installed.
Does that sound good?

Reply:
> I can add a script to carla which prints the libdir. Failing to find/call that script means carla is not installed.

I'd say no.  Reasoning is, I don't think this problem is isolated to Carla and by that I mean if we add more system dependencies, we'll hit the same problems.  As you can see from our `AppRun` script, this isn't the first `LD_LIBRARY_PATH` workaround and I doubt it will be the last.  My opinion is that the `which foo` logic is completely sane.  It should scale well without need to modify the executable, at least not for now.

A better permanent solution I feel is for @probonopd to perhaps look at use-cases like this and help provide sane techniques in the AppImage for optional, system-provided dependencies.  We hit a similar problem with Wine, Jack and now Carla, so this is exposing some edge-cases that I would like to see be an accepted part of the AppImage standard with the ability to both exclude the libraries from being bundled as well as some helper wrappers for setting up the library paths.